web-app/src/main/webapp/script/script.js

```diff
@@ -0,0 +1,138 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+window.onload = async () => {
+  const geocoder = new google.maps.Geocoder();
+  const map = initMap();
+  document
+    .getElementById("report-button")
+    .addEventListener("click", () => showReportForm(map, geocoder));
+  document.getElementById("back-icon").addEventListener("click", () => {
+    hideReportForm();
+    document.getElementById("report-form").reset();
+  });
+  document.getElementById("map-icon").addEventListener("click", () => hideReportForm);
+  document
+    .getElementById("submit-button")
+    .addEventListener("click", () => postUserReport(geocoder));
+  document.getElementById("menu-button").addEventListener("click", () => {
+    document.getElementById("menu").style.display = "block";
+  });
+  document
+    .getElementById("close-menu")
+    .addEventListener("click", () => (document.getElementById("menu").style.display = "none"));
+  
+  const userReports = await fetchAndParseJson("/report");
+  fetchMarkers(map, userReports);
+
+  const timeFrameOptions = document.getElementById("time-frame-options");
+  timeFrameOptions.addEventListener('change', () => { loadPoliceReports(map) });
```

Comment:
I recommend using ES6 `import`s. Otherwise, it's confusing where this `loadPoliceReports` function is coming from. Imports make your dependencies explicit and makes it clear where each symbol is coming from.